### PR TITLE
[SHDOCVW] Initial implementation of Favorites bar

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
First step to Favorites bar support.
Menu: `[View]→[Explorer bar]→[Favorites]`

JIRA issue: [CORE-19686](https://jira.reactos.org/browse/CORE-19686)

## Proposed changes

- Add `CFavBand.cpp`, `CFavBand.h`, `favband.rgs`, and `resource.h` files.
- Modify `CMakeLists.txt`.
- Add `IDS_ADD` and `IDS_ORGANIZE` resource strings.
- Implement empty Favorites bar.
- The treeview is not populated yet.
- The Add button is not implemented yet.

## TODO

- [x] Do tests.

## Screenshot

AFTER:
after

### CODE REVIEW

File: dll/win32/shdocvw/CMakeLists.txt [hunk 1]
```diff
@@ -19,12 +17,23 @@ add_library(shdocvw MODULE
     ${SOURCE}
     mrulist.cpp
     shdocvw.rc
+    $<TARGET_OBJECTS:shdocvw_sublib>
     ${CMAKE_CURRENT_BINARY_DIR}/shdocvw.def)
 
-set_module_type(shdocvw win32dll)
+target_include_directories(shdocvw BEFORE PRIVATE
+    ${REACTOS_SOURCE_DIR}/sdk/include/reactos/wine
+    ${REACTOS_SOURCE_DIR}/sdk/lib/atl)
```
Comment (wjk) [inline dll/win32/shdocvw/CMakeLists.txt:25]: We should use [the `atl_classes` library](https://github.com/reactos/reactos/blob/master/sdk/lib/atl/CMakeLists.txt) instead.

**Review by wjk — CHANGES_REQUESTED** [review]

File: dll/win32/shdocvw/CMakeLists.txt (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shdocvw/CMakeLists.txt:25]:
```suggestion
)
```

File: dll/win32/shdocvw/CFavBand.cpp [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+}
+
+LRESULT CFavBand::OnCommand(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    switch (LOWORD(wParam))
+    {
+        case ID_ADD:
+        {
+            UNIMPLEMENTED;
+            MessageBox(L"Not implemented yet", NULL, MB_ICONERROR);
```
Comment (whindsaks) [inline dll/win32/shdocvw/CFavBand.cpp:229]: Missing translation. You can use the message box in shellutils.h and ERROR_NOT_IMPLEMENTED or another suitable error code.

File: dll/win32/shdocvw/CFavBand.cpp [hunk 3]
```diff
@@ -0,0 +1,648 @@
...
+STDMETHODIMP CFavBand::ResizeBorderDW(const RECT *prcBorder, IUnknown *punkToolbarSite, BOOL fReserved)
+{
+    /* Must return E_NOTIMPL according to MSDN */
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::ShowDW(BOOL fShow)
+{
+    m_fVisible = fShow;
+    ShowWindow(fShow ? SW_SHOWNORMAL : SW_HIDE);
```
Comment (whindsaks) [inline dll/win32/shdocvw/CFavBand.cpp:279]: Just SW_SHOW? Show normal is for top level windows?

File: dll/win32/shdocvw/CFavBand.cpp [hunk 4]
```diff
@@ -0,0 +1,648 @@
...
+    return S_FALSE;
+}
+
+// *** IPersist ***
+
+STDMETHODIMP CFavBand::GetClassID(CLSID *pClassID)
+{
+    if (!pClassID)
+        return E_POINTER;
+    memcpy(pClassID, &CLSID_SH_FavBand, sizeof(CLSID));
```
Comment (whindsaks) [inline dll/win32/shdocvw/CFavBand.cpp:479]: `*pClassID = CLSID_SH_FavBand;`?

File: dll/win32/shdocvw/CFavBand.cpp [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+    DISPPARAMS *pDispParams,
+    VARIANT *pVarResult,
+    EXCEPINFO *pExcepInfo,
+    UINT *puArgErr)
+{
+    switch (dispIdMember)
+    {
+        case DISPID_DOWNLOADCOMPLETE:
+        case DISPID_NAVIGATECOMPLETE2:
+            // FIXME
```
Comment (whindsaks) [inline dll/win32/shdocvw/CFavBand.cpp:603]: What needs fixing here? Can you add a comment?

File: dll/win32/shdocvw/CFavBand.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shdocvw/CFavBand.cpp]: Fixed as `SHELL_ErrorBox(m_hWnd, ERROR_NOT_SUPPORTED);`.

File: dll/win32/shdocvw/CFavBand.cpp (hunk 3 above)
Comment (katahiromz) [reply dll/win32/shdocvw/CFavBand.cpp]: Fixed.

File: dll/win32/shdocvw/CFavBand.cpp (hunk 4 above)
Comment (katahiromz) [reply dll/win32/shdocvw/CFavBand.cpp]: Fixed.

File: dll/win32/shdocvw/CFavBand.cpp (hunk 5 above)
Comment (katahiromz) [reply dll/win32/shdocvw/CFavBand.cpp]: Fixed.

File: dll/win32/shdocvw/CFavBand.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shdocvw/CFavBand.cpp]: error

**Pull request merged**